src/main/java/com/project/controller/ApplicationController.java

```diff
@@ -29,40 +29,24 @@ public ApplicationController(ApplicationService applicationService) {
     }
 
     @GetMapping("api/applications/{id}")
-    public ResponseEntity<Application> getApplication(@PathVariable long id) {
-        try {
-            return new ResponseEntity<Application>(applicationService.getById(id), HttpStatus.OK);
-        } catch (ApplicationNotFoundException e) {
-            return ResponseEntity.notFound().build();
-        }
+    public ResponseEntity<Application> getApplication(@PathVariable long id) throws ApplicationNotFoundException {
```

Comment:
I would suggest changing the service name to ``fetchApplication`` since ``get`` always sounds like a getter from a class.